### PR TITLE
[CX-4731] Middleware workflow action updates

### PR DESCRIPTION
**Pull request by ghost**
Updated this action to depreciated remove 'set-output' function 
Updated several actions causing warnings

Upgraded to node16

Tested branch here: https://github.com/costa-coffee/cex-it-ops-comm/actions/runs/4416792134

### CODE REVIEW

**Review by djdfv — DISMISSED** [review]

**Comment by djdfv** [comment]
@SonnyCostaCoffee Can you fix the tests?

File: dist/index.js
Comment (jayhostan) [inline dist/index.js:1]: did you manually change this?

Comment (jayhostan) [reply dist/index.js:1]: @SonnyCostaCoffee

Comment (ghost) [reply dist/index.js:1]: Yes, seemed to function properly when I tested it.

Comment (ghost) [reply dist/index.js:1]: @jayhostan

**Comment by djdfv** [comment]
Just realised that the test never worked and this is a public repo where we want to reduce exposure as much as possible. We don't even want to expose the name of our secrets in AWS or even set up secrets in a public repo that could be attacked. If Jay is happy, I will approve this PR

File: dist/index.js
Comment (jayhostan) [reply dist/index.js:1]: @SonnyCostaCoffee  you should never do that. Files under /dist are auto-generated, so if anyone updates the code your changes will be lost. You need to build it to update that file.

Comment (ghost) [reply dist/index.js:1]: @jayhostan Updated @action/core and repackaged with node16 

https://github.com/costa-coffee/cex-it-ops-comm/actions/runs/4427825098

**Pull request merged**